File: Yandex/Practicum/Algorightms_and_Data_Structures/4_sprint/A.Кружки/solution.cpp
#include <iostream>
#include <unordered_set>
#include <string>


void solution(int& t) {
    /*
    Time Complexity O(N)
    Memory Complexity O(N)
    */
    std::unordered_set<std::string> hashset;
    for (int i = 0; i < t; ++i) {
        std::string activity;
        std::getline(std::cin, activity);
        if (!hashset.count(activity)) {
            std::cout << activity << "\n";
            hashset.insert(activity);
        }
    }
}


int main() {
    int t;
    std::cin >> t;
    std::cin.ignore();
    solution(t);
}

File: Yandex/Practicum/Algorightms_and_Data_Structures/4_sprint/A.Кружки/solution.py
import sys

def solution() -> None:
    """
    Time Complexity O(N)
    Memory Complexity O(N)
    """
    t: int = int(sys.stdin.readline().rstrip())
    hashset: set[str] = set()
    for _ in range(t):
        activity: str = sys.stdin.readline().rstrip()
        if activity not in hashset:
            hashset.add(activity)
            sys.stdout.write(activity + '\n')

if __name__ == '__main__':
    solution()